src/main/java/edu/harvard/iq/dataverse/MailServiceBean.java

```diff
@@ -164,16 +168,24 @@ public void sendMail(String from, String to, String subject, String messageText)
         sendMail(from, to, subject, messageText, new HashMap<>());
     }
 
-    public void sendMail(String from, String to, String subject, String messageText, Map<Object, Object> extraHeaders) {
+    public void sendMail(String reply, String to, String subject, String messageText, Map<Object, Object> extraHeaders) {
         try {
             MimeMessage msg = new MimeMessage(session);
-            if (from.matches(EMAIL_PATTERN)) {
-                msg.setFrom(new InternetAddress(from));
+            //Always send from system address to avoid email being blocked
+            InternetAddress fromAddress=getSystemAddress();
+            try {
+              fromAddress.setPersonal(fromAddress.getPersonal() + "on behalf of " + reply, charset);
+            } catch (UnsupportedEncodingException ex) {
+                logger.severe(ex.getMessage());
+            }
```

Comment:
We're catching `UnsupportedEncodingException` here and logging the exception, which is good, but should we return in this `catch` rather than marching forward?

Reply:
I was mostly following the pattern in the other methods, but this exception would be non-fatal (and presumably not likely since charset is set to 'UTF-8' on line 73). The message could go out without the 'on behalf of <x>' addition, which would probably be better (?) than failing to send anything.